### PR TITLE
Use modern ES syntax rather than `.call`

### PR DESCRIPTION
**Pull request by renchap**
I replaced `Object.prototype.hasOwnProperty.call` with the new ES2022 `Object.hasOwn`.

It is not transpiled by Babel in the final files, because this function is supported by all the browsers our BrowsersList is targetting.

If this causes issues, then we can change our supported browsers list and target earlier versions, then it will be polyfilled accordingly.

### CODE REVIEW

**Review by ykzts — DISMISSED** [review]

File: app/javascript/packs/admin.jsx [hunk 1]
```diff
@@ -219,7 +219,7 @@ ready(() => {
   });
 
   Rails.delegate(document, 'form', 'submit', ({ target }) => {
-    [].forEach.call(target.querySelectorAll('input[type="datetime-local"]'), element => {
+    target.querySelectorAll('input[type="datetime-local"]').foreach(element => {
```
Comment (ClearlyClaire) [inline app/javascript/packs/admin.jsx:222]: Typo:
```suggestion
    target.querySelectorAll('input[type="datetime-local"]').forEach(element => {
```

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]
Typo, plus `some` and `every` are not defined on `NodeList` afaik

File: app/javascript/packs/public.jsx [hunk 2]
```diff
@@ -128,8 +128,8 @@ function loaded() {
   if (reactComponents.length > 0) {
     import(/* webpackChunkName: "containers/media_container" */ '../mastodon/containers/media_container')
       .then(({ default: MediaContainer }) => {
-        [].forEach.call(reactComponents, (component) => {
-          [].forEach.call(component.children, (child) => {
+        reactComponents.forEach((component) => {
+          component.children.forEach((child) => {
```
Comment (ClearlyClaire) [inline app/javascript/packs/public.jsx:132]: `component.children` is a `HTMLCollection`, which doesn't have `forEach`

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]

**Review by ClearlyClaire — APPROVED** [review]
Looks good now, although I'd like to see it run with the test introduced in #29424 first.

**Comment by renchap** [comment]
Rebased to include your new test

**Pull request merged**